validator/delphi_validator/validate.py

```diff
@@ -0,0 +1,742 @@
+                    ("check_val_pct_gt_100", nameformat),
+                    df_to_test[(df_to_test['val'] > 100)],
+                    "val column can't have any cell greater than 100 for percents"))
+
+        if proportion_option:
+            if not df_to_test[(df_to_test['val'] > 100000)].empty:
+                self.raised_errors.append(ValidationError(
+                    ("check_val_prop_gt_100k", nameformat),
+                    df_to_test[(df_to_test['val'] > 100000)],
+                    "val column can't have any cell greater than 100000 for proportions"))
+
+        if df_to_test['val'].isnull().values.any():
+            self.raised_errors.append(ValidationError(
+                ("check_val_missing", nameformat),
+                None, "val column can't have any cell that is NA"))
+
+        if not df_to_test[(df_to_test['val'] < 0)].empty:
+            self.raised_errors.append(ValidationError(
+                ("check_val_lt_0", nameformat),
+                df_to_test[(df_to_test['val'] < 0)],
+                "val column can't have any cell smaller than 0"))
+
+    def check_bad_se(self, df_to_test, nameformat):
+        """
+        Check standard errors for validity.
+
+        Arguments:
+            - df_to_test: pandas dataframe of a single CSV of source data
+            (one day-signal-geo_type combo)
+            - nameformat: str CSV name; for example, "20200624_county_smoothed_nohh_cmnty_cli.csv"
+
+        Returns:
+            - None
+        """
+        # Add a new se_upper_limit column.
+        df_to_test.eval(
+            'se_upper_limit = (val * sample_size + 50)/(sample_size + 1)', inplace=True)
+
+        df_to_test['se'] = df_to_test['se'].round(3)
+        df_to_test['se_upper_limit'] = df_to_test['se_upper_limit'].round(3)
+
+        if not self.missing_se_allowed:
+            # Find rows not in the allowed range for se.
+            result = df_to_test.query(
+                '~((se > 0) & (se < 50) & (se <= se_upper_limit))')
+
+            if not result.empty:
+                self.raised_errors.append(ValidationError(
+                    ("check_se_not_missing_and_in_range", nameformat),
+                    result, "se must be in (0, min(50,val*(1+eps))] and not missing"))
+
+            if df_to_test["se"].isnull().mean() > 0.5:
+                self.raised_errors.append(ValidationError(
+                    ("check_se_many_missing", nameformat),
+                    None, 'Recent se values are >50% NA'))
+
+        elif self.missing_se_allowed:
+            result = df_to_test.query(
+                '~(se.isnull() | ((se > 0) & (se < 50) & (se <= se_upper_limit)))')
+
+            if not result.empty:
+                self.raised_errors.append(ValidationError(
+                    ("check_se_missing_or_in_range", nameformat),
+                    result, "se must be NA or in (0, min(50,val*(1+eps))]"))
+
+        result_jeffreys = df_to_test.query('(val == 0) & (se == 0)')
```

Comment:
Is "jefferys" an established term for this?

Reply:
Yes; see [prior work](https://github.com/cmu-delphi/covidcast-indicators/issues/54) and [documentation]9https://cmu-delphi.github.io/delphi-epidata/api/covidcast-signals/hospital-admissions.html); ["Jeffreysized"](https://cmu-delphi.github.io/delphi-epidata/api/covidcast-signals/indicator-combination.html) is also used